build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```
@@ -212,12 +212,7 @@ fun Test.configureJvmForTest() {
     jvmArgumentProviders.add(CiEnvironmentProvider(this))
     val launcher = project.javaToolchains.launcherFor {
         languageVersion.set(jvmVersionForTest())
-        project.testJavaVendor.map {
-            when (it.toLowerCase()) {
-                "oracle" -> vendor.set(JvmVendorSpec.ORACLE)
-                "openjdk" -> vendor.set(JvmVendorSpec.ADOPTIUM)
-            }
-        }.getOrNull()
+        vendor.set(project.testJavaVendor.orNull)
```

```suggestion
        vendor.set(project.testJavaVendor)
```

Why the `orNull` here?

---

Using `orNull` forces `vendor.set(JvmVendorSpec)` is selected. Without `orNulll`, `vendor.set(Provider<JvmVendorSpec>)` is selected and there will be error if it's not set:

```
> Error while evaluating property 'javaLauncher' of task ':cli:test'
   > Failed to query the value of task ':cli:test' property 'javaLauncher'.
      > Cannot query the value of property 'vendor' because it has no value available.

```

---

Ah, that is probably https://github.com/gradle/gradle/issues/14768. So if there is nothing passed in on the command line, we'd fall back on the `ANY` vendor, though that is currently not public API :(.